### PR TITLE
GUVNOR-3432: [Guided Decision Table] Formulas and Predicates cannot be set

### PR DESCRIPTION
**Pull request by karreiro**
See https://issues.jboss.org/browse/GUVNOR-3432

Before:
before

After:
after

### CODE REVIEW

**Comment by jomarko** [comment]
jenkins execute full downstream build

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/wizard/column/pages/FieldPageTest.java [hunk 1]
```diff
@@ -374,22 +374,75 @@ public void testAsWidget() {
...
+
+        doReturn(factType).when(page).factType();
+
+        page.setupPatternWarningMessages();
+
+        verify(view).patternWarningToggle(false);
+    }
+
+    @Test
+    public void testSetupPatternWarningMessagesWhenFactTypeIsNilAndConstraintValueIsPredicate() {
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/wizard/column/pages/FieldPageTest.java]: @karreiro This test is great, it brought me to this observation:
- If Calculation Type is Predicate, Field Page doesn't force the user to specify pattern
- However, the column wizard always force the user to specify pattern on the pattern page
- Second however, I am not sure, how to solve this

Comment (manstis) [reply drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/wizard/column/pages/FieldPageTest.java]: A Pattern always needs to be selected; therefore IMO even if the Calculation Type is Predicate the warning about Pattern not being selected should be shown on the Field page. When using a Predicate it is not possible to complete the wizard unless a Pattern has been selected.

Comment (jomarko) [reply drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/wizard/column/pages/FieldPageTest.java]: Thanks Michael, +1 from my site. Guilherme, please incorporate also this.

**Comment by karreiro** [comment]
Changes applied. Thank you for the review @manstis and @jomarko.

---

demo

**Comment by karreiro** [comment]
Jenkins please execute full downstream build.

**Pull request merged**